File: tasks/kthDivisor.py
def kthDivisor(n, k):
    divisors = []
    i = 1
    while i * i < n:
        if n % i == 0:
            divisors.append(i)
        i += 1
    if i * i > n:
        i -= 1
    while i > 0:
        if n % i == 0:
            divisors.append(n/i )
        i -= 1
    if len(divisors) < k:
        return -1
    return divisors[k - 1]

File: tasks/kthDivisor.java
int kthDivisor(int n, int k) {
  ArrayList<Integer> divisors = new ArrayList<>();
  int i = 1;
  while (i * i < n) {
    if (n % i == 0) {
      divisors.add(i);
    }
    i++;
  }
  if (i * i > n) {
    i--;
  }
  while (i > 0) {
    if (n % i == 0) {
      divisors.add(n/i);
    }
    i--;
  }
  if (divisors.size() < k) {
    return -1;
  }
  return divisors.get(k - 1);
}
